applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -361,58 +394,34 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
                 pScheme->CalculateSystemContributions(*(it_el.base()), LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
                 Element::DofsVectorType dofs;
                 it_el->GetDofList(dofs, CurrentProcessInfo);
-                //assemble the elemental contribution - here is where the ROM acts
-                //compute the elemental reduction matrix Phi
                 const auto &geom = it_el->GetGeometry();
-                Matrix PhiElemental(geom.size() * mNodalDofs, mRomDofs);
-
-                for (unsigned int i = 0; i < geom.size(); ++i){
-                    const Matrix &rom_nodal_basis = geom[i].GetValue(ROM_BASIS);
-                    for (unsigned int k = 0; k < rom_nodal_basis.size1(); ++k){
-                        if (dofs[i * mNodalDofs + k]->IsFixed())
-                            row(PhiElemental, i * mNodalDofs + k) = ZeroVector(PhiElemental.size2());
-                        else
-                            row(PhiElemental, i * mNodalDofs + k) = row(rom_nodal_basis, k);
-                    }
-                }
+                Matrix PhiElemental(dofs.size(), mRomDofs);

```

Comment:
allocate this matrix outside the loop (as a matrix of size 0*0) and pass it as firstprivate. It will be resized as needed within the loop, however you will not need to allocate a new one for every element!

Reply:
Added this one, but the OpenMP parallelization will be added right after aproval